data/basics/userBalance.go

```diff
@@ -153,6 +185,63 @@ type AccountData struct {
 	// A transaction may change an account's AuthAddr to "re-key" the account.
 	// This allows key rotation, changing the members in a multisig, etc.
 	AuthAddr Address `codec:"spend"`
+
+	// AppLocalStates stores the local states associated with any applications
+	// that this account has opted in to.
+	AppLocalStates map[AppIndex]AppLocalState `codec:"appl,allocbound=encodedMaxAppLocalStates"`
+
+	// AppParams stores the global parameters and state associated with any
+	// applications that this account has created.
+	AppParams map[AppIndex]AppParams `codec:"appp,allocbound=encodedMaxAppParams"`
+
+	// TotalAppSchema stores the sum of all of the LocalStateSchemas
+	// and GlobalStateSchemas in this account (global for applications
+	// we created local for applications we opted in to), so that we don't
+	// have to iterate over all of them to compute MinBalance.
+	TotalAppSchema StateSchema `codec:"tsch"`
+}
+
+// AppLocalState stores the LocalState associated with an application. It also
+// stores a cached copy of the application's LocalStateSchema so that
+// MinBalance requirements may be computed 1. without looking up the
+// AppParams and 2. even if the application has been deleted
+type AppLocalState struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	Schema   StateSchema  `codec:"hsch"`
+	KeyValue TealKeyValue `codec:"tkv"`
+}
+
+// AppParams stores the global information associated with an application
+type AppParams struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	ApprovalProgram   []byte      `codec:"approv"`
```

Comment:
this need to be allocbound. ( yes, I know that we currently don't support byte slices allocbounding )

Reply:
I don't want to add an `allocbound` if `msgp` doesn't support it, otherwise I think it is confusing to someone reading the code.